File: LeetCode/530-minimum-absolute-difference-in-bst/ex_530.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    int min = 0x7fffffff;
    int prev = -1;
    public int getMinimumDifference(TreeNode root) {
        if (root == null)   return min;
        getMinimumDifference(root.left);
        if (prev != -1)     min = Math.min(min, root.val - prev);
        prev = root.val;
        getMinimumDifference(root.right);
        return min;
    }
}

File: LeetCode/530-minimum-absolute-difference-in-bst/ex_530.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    last = -1
    diff = 0x7FFFFFFF
    def getMinimumDifference(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        if not root:    return self.diff
        self.getMinimumDifference(root.left)
        if self.last != -1:
            self.diff = min(self.diff, root.val - self.last)
        self.last = root.val
        self.getMinimumDifference(root.right)
        return self.diff